### ISSUE TITLE
Model bugs out when baking shape keys, then deleting helpers (Attempting to add Shape Keys to human)

### ISSUE THREAD

**Issue by ValiantShuffle** [makehumancommunity/mpfb2#98]
**Describe the bug**
I don't know if I should put this as a support request, or a bug, because I could be doing something completely wrong.

I followed [this guide](https://github.com/makehumancommunity/mpfb2/issues/71#issuecomment-1358890124) because setting up sculpting through [this video](https://youtu.be/9jmTdhVjAsI?t=872) didn't work the way I needed it to work. I have a human that I'd like to add a lot of Shape Keys to, so they can morph into different types of things, or different physiques.

I have a model that has their bones (Default with toes), high-poly eyes, eyebrows, and skin set up, then i followed the link's instructions to make shape keys work, while (hopefully) keeping the model intact. Once you finish step 1, the model can become all sorts of wacky things. The more you edit the rig's bones, the crazier things can end up.

Basically what it's seeming like I have to do, is I have to take the model out of mpfb2 plugin, and add onto it without breaking things. 

Is there any way I can make new shape keys without running into this issue?

**Screenshots / logs**
https://drive.proton.me/urls/QSYB09X53W#5iMmaQcs4XDr 

**Platform and versions**
- SteamOS 3.4.6 (Arch Linux)
- Blender 3.5.0 (Flatpak)
- mpfb2-20230412 (Downloaded about 16hrs ago, along with the CC0 Asset Pack)

(Sorry if this topic was too similar to the previously linked Github issue, I wasn't sure where to put this as this as I've never had a Github account until today)

**Comment by dodopod** [comment]
This is happening for me too, except I am setting up sculpting like in the video.

@ValiantShuffle Try removing the "Armature PV" modifier. That fixed it for me, for some reason.

**Comment by joepal1976** [comment]
I'm not entirely sure I understand what you are trying to achieve, but adding new shape keys is actually a typical use case for MPFB (and makehuman).

For new shapekeys to play nice with the other shapekeys, you will need to take a few things into account.

First: new shape keys should be made with MakeTarget (under "create assets"). If you set up a character as a starting point and "apply shape keys", you can then create a new target from that. The created target file can be added as a custom target which makes it end up as a slider on the modeling tab.

Second: armature fitting is heavily dependent on joint cubes. These are hidden geometry which you can see if you toggle the mask modifier on the character. If you modify the position of a joint (for example an elbow), then be sure to also move the joint cube which is inside the elbow. Otherwise the rigging will cause strange results.

While this might seem a bit cumbersome compared to just going ahead and adding shapekeys directly to an existing character, it has the upside that the shape key can then be re-used on other characters.

**Comment by ValiantShuffle** [comment]
> I'm not entirely sure I understand what you are trying to achieve, but adding new shape keys is actually a typical use case for MPFB (and makehuman).
> 
> For new shapekeys to play nice with the other shapekeys, you will need to take a few things into account.
> 
> First: new shape keys should be made with MakeTarget (under "create assets"). If you set up a character as a starting point and "apply shape keys", you can then create a new target from that. The created target file can be added as a custom target which makes it end up as a slider on the modeling tab.
> 
> Second: armature fitting is heavily dependent on joint cubes. These are hidden geometry which you can see if you toggle the mask modifier on the character. If you modify the position of a joint (for example an elbow), then be sure to also move the joint cube which is inside the elbow. Otherwise the rigging will cause strange results.
> 
> While this might seem a bit cumbersome compared to just going ahead and adding shapekeys directly to an existing character, it has the upside that the shape key can then be re-used on other characters.

Sorry for confusing you, basically what I'm attempting to do is to create a "god" that takes on different body types, and can also transform into different creatures. I know for the body types, I can do this through shape keys to make it a lot easier on myself, I just didn't know how to get to that point with mpfb2 (I'm not good at modeling humans yet, so having a human I could already work with is an amazing baseline. Thank you and other contributors for helping me with that!).

Originally, I thought I'd have to take the model out of mpfb2 and just work without the plugin to get the model working the way I wanted it to. It's why I followed the previously linked github issue because it was the closest thing I could have found that would get it kinda close to how I wanted it to work (excluding the armature issues).

I couldn't get the mask modifier to properly work, but I don't think I'll need it yet (I just have the IK helpers, using those instead of individually moving bones) considering my character moves perfectly fine inside the plugin.

The benefits outweigh the downsides (if you even really wanna call it cumbersome, imo it really isn't, it has massive benefits because it becomes way easier to import characters to other projects using mpfb2, and the ability to one and done this is way too good)

Thank you for your help! It works, and I'll continue to use it as time goes on!
